src/main/java/amidst/gui/profileselect/LocalProfileComponent.java

```diff
@@ -78,9 +80,16 @@ public void load() {
 
 	@CalledOnlyBy(AmidstThread.WORKER)
 	private boolean tryLoad() {
+		// TODO: Replace with proper handling for modded profiles.
 		try {
-			Log.i("using minecraft launcher profile '" + getProfileName() + "' with versionId '" + getVersionName()
-					+ "'");
+			Log.i("using minecraft launcher profile '" + getProfileName() + "' with versionId '" + getVersionName() + "'");
+
+			String possibleModProfiles = ".*[optifine|forge].*";
+			if (Pattern.matches(possibleModProfiles, getVersionName().toLowerCase(Locale.ENGLISH))) {
```

Comment:
This is still broken. `[]` is used to represent a set of characters. For example `[hello]` means *a single character from the set {'h', 'e', 'l', 'o'}*.

Please use the following statement:

```
Pattern.compile("optifine|forge").matcher(getVersionName().toLowerCase(Locale.ENGLISH)).find()
```

If you want to, you can extract `Pattern.compile("optifine|forge")` as a constant.